lib/apiservers/engine/backends/archive.go

```diff
@@ -107,26 +100,24 @@ func (c *Container) ContainerExport(name string, out io.Writer) error {
 // be an error if unpacking the given content would cause an existing directory
 // to be replaced with a non-directory and vice versa.
 func (c *Container) ContainerExtractToDir(name, path string, noOverwriteDirNonDir bool, content io.Reader) error {
-	err := fmt.Errorf("%s does not yet implement ContainerExtractToDir", ProductName())
+	defer trace.End(trace.Begin(name))
 
-	// defer trace.End(trace.Begin(name))
-
-	// vc := cache.ContainerCache().GetContainer(name)
-	// if vc == nil {
-	// 	return NotFoundError(name)
-	// }
+	vc := cache.ContainerCache().GetContainer(name)
+	if vc == nil {
+		return NotFoundError(name)
+	}
 
-	// err := c.importToContainer(vc, path, content)
-	// if err != nil && IsResourceInUse(err) {
-	// 	log.Errorf("ContainerExtractToDir failed, resource in use: %s", err.Error())
+	err := c.importToContainer(vc, path, content)
+	if err != nil && IsResourceInUse(err) {
+		log.Errorf("ContainerExtractToDir failed, resource in use: %s", err.Error())
```

Comment:
Since this is likely to be caused by another person/script issuing commands( primarily `docker stop/start`). We should also let the user know that an additional cp call should succeed unless the resource is being thrashed(multiple actors calling start/stop repeatedly). Maybe just add on, "Please reissue the command".  -emma,zach,matt